int/api/cmd/execute_function.go

```diff
@@ -13,40 +11,37 @@ import (
 	"github.com/massalabs/thyra/pkg/wallet"
 )
 
-func NewExecuteFunction(walletStorage *sync.Map) *FunctionExecuter {
-	return &FunctionExecuter{walletStorage: walletStorage}
+func NewExecuteFunction() *FunctionExecuter {
+	return &FunctionExecuter{}
 }
 
-type FunctionExecuter struct {
-	walletStorage *sync.Map
-}
+type FunctionExecuter struct{}
 
 //nolint:nolintlint,ireturn
 func (f *FunctionExecuter) Handle(params operations.CmdExecuteFunctionParams) middleware.Responder {
 	addr, err := base58.CheckDecode((*params.Body.At)[1:])
 	if err != nil {
-		panic(err)
+		return operations.NewCmdExecuteFunctionUnprocessableEntity().WithPayload(
+			&models.Error{Code: errorCodeUnknownKeyID, Message: "Error: cannot decode Smart contract address : " + err.Error()})
 	}
 
 	addr = addr[1:]
 
-	value, exists := f.walletStorage.Load(*params.Body.Name)
-	if !exists {
+	wallet, err := wallet.Load(*params.Body.Nickname)
+	if err != nil {
 		return operations.NewCmdExecuteFunctionUnprocessableEntity().WithPayload(
 			&models.Error{
 				Code:    errorCodeUnknownKeyID,
-				Message: "Error: unknown wallet nickname : " + *params.Body.Name,
+				Message: "Error: unknown wallet nickname : " + *params.Body.Nickname,
```

Comment:
How do you know the wallet is unknown ?
wallet.Load returns two kind of error:
- reading file error
- parsing file error

I think the best way to know the error is to return the error :-)